new_dataloader.py

```diff
@@ -237,34 +281,41 @@ def label2onehot(self, labels, dim):
         return out.float() 
        
     def process(self, size= None):
-        
-        mols = [Chem.MolFromSmiles(line) for line in open(self.raw_files, 'r').readlines()]
-     
-        mols = list(filter(lambda x: x.GetNumAtoms() <= self.max_atom, mols))
-        mols = mols[:size]
-        indices = range(len(mols))
-        
-        self._generate_encoders_decoders(mols)
-        
-  
+        '''
+        Process the dataset. This function will be only run if processed_file_names does not exist in the data folder already.
+        '''
+        # mols = [Chem.MolFromSmiles(line) for line in open(self.raw_files, 'r').readlines()]
+        # mols = list(filter(lambda x: x.GetNumAtoms() <= self.max_atom, mols))
+        # mols = mols[:size] # i 
+        # indices = range(len(mols))
+
+        smiles = pd.read_csv(self.raw_files, header=None)[0].tolist()
+        self._generate_encoders_decoders(smiles)
     
-        pbar = tqdm(total=len(indices))
-        pbar.set_description(f'Processing chembl dataset')
-        max_length = max(mol.GetNumAtoms() for mol in mols)
+        # pbar.set_description(f'Processing chembl dataset')
+        # max_length = max(mol.GetNumAtoms() for mol in mols)
```

Comment:
Commenting out this line means that we don't get the advantage of lower computation advantage in small molecule datasets. But also calculating this for every dataset will take extra time during every run.

However, we can still achieve this by adding this to the for loop in `_generate_encoders_decoders`:
```
max_atom_size = 0
for smile in data:
    mol = Chem.MolFromSmiles(smile)
    atom_labels.update([atom.GetAtomicNum() for atom in mol.GetAtoms()])
    # bond_types.update([bond.GetBondType() for bond in mol.GetBonds()])
    max_atom_size = max(max_atom_size, mol.GetNumAtoms())
print("Max atom size", max_atom_size)
```

Reply:
Well, adding this computation resulted in 7 minute shorter runtime. :) 

It found max atom size is 548 in training, and 499 in test data. 
Obviously we can should find minimum of this value and max_length value